### PR TITLE
base64_decode opcode cost

### PR DESCRIPTION
**Pull request by sczembor**
Closes #x

## Proposed Changes

+ Update the `execute` method to return the cost 
+ Add unit tests for the method
+ Update the rest of the opcodes to return their cost from `execute` method (Only the opcodes that cost is different from 1 return their cost

## Potential followups
+ Add testing for the interpreter to check if the the cost is correct

### CODE REVIEW

File: packages/runtime/src/interpreter/interpreter.ts [hunk 1]
```diff
@@ -548,14 +545,13 @@ export class Interpreter {
 					tealVersion: this.tealVersion,
 				});
 			}
+			const costFromExecute = instruction.execute(this.stack);
+			const cost =
+				costFromExecute === undefined ? this.lineToCost[instruction.line] : costFromExecute;
```
Comment (vuvth) [inline packages/runtime/src/interpreter/interpreter.ts:550]:
```suggestion
			const costFromExecute = instruction.execute(this.stack);
			const cost = costFromExecute ?? this.lineToCost[instruction.line];
```

Comment (vuvth) [reply packages/runtime/src/interpreter/interpreter.ts:550]: Should we return cost with all `instruction.execute()` ? Maybe we should do it in another PR.

File: packages/runtime/src/interpreter/opcode-list.ts [hunk 2]
```diff
@@ -4794,9 +4794,10 @@ export class Base64Decode extends Op {
 		}
 	}
 
-	execute(stack: TEALStack): void {
+	execute(stack: TEALStack): number {
 		this.assertMinStackLen(stack, 1, this.line);
 		const last = this.assertBytes(stack.pop(), this.line);
+		const cost = 1 + Math.ceil(last.length / 16);
```
Comment (vuvth) [inline packages/runtime/src/interpreter/opcode-list.ts:4800]: Please add comment to explain compute `cost` formula.

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 3]
```diff
@@ -6667,6 +6667,41 @@ describe("Teal Opcodes", function () {
...
+			assert.deepEqual(5, cost); // base64_decode cost = 5 (60 bytes -> 1 + ceil(60/16))
+
+			toPush = Buffer.from(
+				"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_AA==",
+				"utf-8"
+			);
+			stack.push(toPush);
+			op = new Base64Decode(["URLEncoding"], 0);
+			cost = op.execute(stack);
+			assert.deepEqual(6, cost); // base64_decode cost = 6 (68 bytes -> 1 + ceil(68/16))
```
Comment (vuvth) [inline packages/runtime/test/src/interpreter/opcode-list.ts:6702]: How about the case `base64_opcode StdEncoding`?

Comment (vuvth) [reply packages/runtime/test/src/interpreter/opcode-list.ts:6702]: Ah, this test case for compute cost. Thanks!

**Review by vuvth — APPROVED** [review]
Nice work! I felt few comments.

File: packages/runtime/src/interpreter/interpreter.ts (hunk 1 above)
Comment (sczembor) [reply packages/runtime/src/interpreter/interpreter.ts:550]: I will do it in next PR

File: packages/runtime/src/interpreter/opcode-list.ts (hunk 2 above)
Comment (sczembor) [reply packages/runtime/src/interpreter/opcode-list.ts]: Done

File: packages/runtime/src/interpreter/interpreter.ts (hunk 1 above)
Comment (sczembor) [reply packages/runtime/src/interpreter/interpreter.ts]: I added the cost for the rest of the opcodes

**Pull request merged**